snapcraft/internal/remote_build/_launchpad.py

```diff
@@ -0,0 +1,233 @@
+        except restfulclient.errors.NotFound:
+            pass
+
+    def start_build(self, timeout: int = 5, attempts: int = 5) -> int:
+        """Initiate a new snap build."""
+        owner = self._lp.people[self.user]
+        dist = self._lp.distributions["ubuntu"]
+        snap = self._lp.snaps.getByName(name=self._id, owner=owner)
+        snap_build_request = snap.requestBuilds(
+            archive=dist.main_archive,
+            channels={"core": self._core_channel, "snapcraft": self._snapcraft_channel},
+            pocket="Updates",
+        )
+        logger.debug("Request URL: {}".format(snap_build_request))
+
+        build_number = snap_build_request.self_link.rsplit("/", 1)[-1]
+
+        for i in range(0, attempts):
+            builds = self._lp.load(snap_build_request.builds_collection_link)
+            logger.debug(
+                "Builds collection entries: {} ({})".format(len(builds.entries), i)
+            )
+            if builds.entries:
+                break
+            time.sleep(timeout)
+        else:
+            self.delete_snap()
+            raise RemoteBuilderNotReadyError()
+
+        self._waiting = [build["arch_tag"] for build in builds.entries]
+        self._builds_collection_link = snap_build_request.builds_collection_link
+
+        return build_number
+
+    def recover_build(self, req_number: int) -> None:
+        """Prepare internal state to monitor an existing build."""
+        url = "https://api.launchpad.net/devel/~{}/+snap/{}/+build-request/{}".format(
+            self.user, self._id, req_number
+        )
+
+        try:
+            request = self._lp.load(url)
+        except restfulclient.errors.NotFound:
+            raise RemoteBuildNotFoundError(name=self._name, req_number=req_number)
+
+        logger.debug("Request URL: {}".format(request))
+        builds = self._lp.load(request.builds_collection_link)
+        self._waiting = [build["arch_tag"] for build in builds.entries]
+        self._builds_collection_link = request.builds_collection_link
+
+    def monitor_build(self, interval: int = _LP_POLL_INTERVAL) -> None:
+        """Check build progress, and download artifacts when ready."""
+        logger.debug("Monitoring builds: {}".format(" ".join(self._waiting)))
+        while len(self._waiting):
+            time.sleep(interval)
+            builds = self._lp.load(self._builds_collection_link)
+            if not builds.entries:
+                break
+
+            for build in builds.entries:
+                arch = build["arch_tag"]
+                build_state = build["buildstate"]
+                logger.debug("{} state: {}".format(arch, build_state))
+                if arch in self._waiting:
+                    if build_state == _LP_SUCCESS_STATUS:
+                        self._process_build(build)
+                    elif build_state == _LP_FAIL_STATUS:
+                        self._process_fail(build)
+
+    def show_build_status(self) -> None:
```

Comment:
can this be "get_build_status" and have the user of this method print it instead?

Reply:
Done.